netket/graph/lattice.py

```diff
@@ -104,323 +107,349 @@ def get_edges(atoms_positions, cutoff, distance_atol=cutoff_tol):
     return pairs
 
 
-def create_points(basis_vectors, extent, atom_coords, pbc):
+@dataclass
+class LatticeSite:
+    id: int
+    """Integer ID of this site"""
+    position: Array
+    """Real-space position of this site"""
+    cell_coord: Array
+    """Cell coordinates of this site"""
+    inside: bool
+    """TODO: When exactly is this needed?"""
+
+    def __repr__(self):
+        s = ", ".join(map(str, (self.id, self.cell_coord)))
+        return f"LatticeSite({s})"
+
+
+def create_sites(primitives, extent, apositions, pbc):
     shell_vec = _np.zeros(extent.size, dtype=int)
     shift_vec = _np.zeros(extent.size, dtype=int)
     # note: by modifying these, the number of shells can be tuned.
     shell_vec[pbc] = 2
     shift_vec[pbc] = 1
     ranges = tuple([list(range(ex)) for ex in extent + shell_vec])
-    atoms = []
-    cellANDlabel_to_site = {}
+    sites = []
+    cell_coord_to_site = {}
     for s_cell in itertools.product(*ranges):
         s_coord_cell = _np.asarray(s_cell) - shift_vec
         if _np.any(s_coord_cell < 0) or _np.any(s_coord_cell > (extent - 1)):
             inside = False
         else:
             inside = True
```

Comment:
I don't think I've seen a case where this was not `True` so I am unsure in which case `inside` (which is stored for all sites) is actually needed.

Reply:
I think this only exists as an attempt to ensure basis sites are not duplicated. There are better ways to achieve that (e.g. running the "hashing trick" on fractional coordinates), so I plan to chuck this sort of thing out when refactoring `Lattice`

Reply:
Also, this holds for all our lattices in the tests (which is relatively representative of physically relevant lattices):
```python
def test_all_inside():
    for lattice in graphs:
        if not isinstance(lattice, Lattice):
            continue
        assert all(s.inside is True for s in lattice.sites)
```
So probably we can get away with not storing it indeed. Maybe you can remove it from `LatticeSite` when you are refactoring the code using it.

Reply:
This is something I intend to chuck too when I refactor `_clean_basis`. It is an unnecessary restriction since the lookup uses periodic boundary conditions correctly, and it is easy to check whether two sites are shifted by some lattice vector using a variant of the same logic

Reply:
Sound good. So I'll just leave this as is for now and you can change it later.